apps/client/src/services/socket-communication.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  PROTO_AUTH_TOKEN_TRANSFER,
+  PROTO_CREATE_LOBBY_ACCEPT,
+  PROTO_CREATE_LOBBY_REQUEST,
+  PROTO_JOIN_LOBBY_ACCEPT,
+  PROTO_JOIN_LOBBY_REQUEST,
+  RECONNECT_WAITING_TIME,
+  REQUEST_WAITING_TIME,
+} from '@razor/constants';
+import {
+  AuthToken,
+  InitialClientData,
+  InitialServerData,
+  playerIdSchema,
+  stateModelSchema,
+  tournamentIdSchema,
+} from '@razor/models';
+import { io, Socket } from 'socket.io-client';
+
+import { PubSubEvents } from '../models';
+import { pubsub } from '../utils/pubsub';
+
+const SOCKET_ENDPOINT =
+  process.env['NX_SOCKET_ENDPOINT'] || 'http://localhost:3000';
+let authToken = '';
+let savedPlayerName = '';
+let savedRoomId = '';
+
+interface SocketFormat extends Socket {
+  auth: {
+    token: AuthToken;
+  };
+}
+
+const socket = io(SOCKET_ENDPOINT, {
+  auth: {
+    token: authToken,
+  },
+  autoConnect: false,
+  withCredentials: true,
+}) as SocketFormat;
+
+export const endSocket = (): void => {
+  socket.disconnect();
+};
+
+const initializeSocket = (): void => {
+  socket.auth.token = authToken;
+  socket.connect();
+  socket.on('connect_error', () => {
+    alert('Connection error');
+    endSocket();
+  });
+  socket.on('connect', () => {
+    console.log('connected');
+  });
+  socket.on(PROTO_AUTH_TOKEN_TRANSFER, (token: string) => {
+    authToken = token;
+  });
+};
+
+socket.on('disconnect', reason => {
+  console.log('disconnected', reason, savedRoomId, savedPlayerName);
```

Comment:
Shall we extract this to a separate function without inlining it

Reply:
Fixed 22b751787985f56dd1197d9620c9f86d6ae43aca PR #81